src/main/java/org/truffleruby/core/kernel/KernelNodes.java

```diff
@@ -1789,6 +1789,21 @@ public DynamicObject toS(Object self) {
 
     }
 
+    @Primitive(name = "object_to_s")
+    public abstract static class ObjectToSNode extends PrimitiveArrayArgumentsNode {
+
+        @Specialization
+        public DynamicObject toS(Object obj,
+                                 @Cached("createKernelToSNode()") ToSNode kernelToSNode) {
```

Comment:
Could you add a `create()` method on ToSNode instead?
Then other usages can also use it.

Reply:
Fixed, thanks.

Related to this, I think it would be nice if `@CoreMethod` had a `primitive` attribute allowing them to be invoked  like primitive nodes. E.g. `@CoreMethod(names = {"to_s", "inspect"}, primitive = "object_to_s")`